transactions/self_payment_detection_test.go

```diff
@@ -0,0 +1,214 @@
+}
+
+func TestSendPaymentSync_SelfPaymentDetection_MultipleIncomingTransactions(t *testing.T) {
+	ctx := context.TODO()
+
+	svc, err := tests.CreateTestService(t)
+	require.NoError(t, err)
+	defer svc.Remove()
+
+	svc.LNClient.(*tests.MockLn).Pubkey = "03cbd788f5b22bd56e2714bff756372d2293504c064e03250ed16a4dd80ad70e2c"
+
+	mockPreimage := "123preimage"
+	svc.DB.Create(&db.Transaction{
+		State:          constants.TRANSACTION_STATE_SETTLED,
+		Type:           constants.TRANSACTION_TYPE_INCOMING,
+		PaymentRequest: tests.MockInvoice,
+		PaymentHash:    tests.MockPaymentHash,
+		Preimage:       &mockPreimage,
+		AmountMsat:     123000,
+	})
+	svc.DB.Create(&db.Transaction{
+		State:          constants.TRANSACTION_STATE_PENDING,
+		Type:           constants.TRANSACTION_TYPE_INCOMING,
+		PaymentRequest: tests.MockInvoice,
+		PaymentHash:    tests.MockPaymentHash,
+		Preimage:       &mockPreimage,
+		AmountMsat:     123000,
+	})
+
+	transactionsService := NewTransactionsService(svc.DB, svc.EventPublisher)
+	transaction, err := transactionsService.SendPaymentSync(ctx, tests.MockInvoice, nil, nil, svc.LNClient, nil, nil, nil)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, transaction)
+	assert.True(t, transaction.SelfPayment)
+	assert.Equal(t, constants.TRANSACTION_STATE_SETTLED, transaction.State)
+}
+
+func TestSendPaymentSync_SelfPaymentDetection_OutgoingTransactionOnly(t *testing.T) {
+	ctx := context.TODO()
+
+	svc, err := tests.CreateTestService(t)
+	require.NoError(t, err)
+	defer svc.Remove()
+
+	svc.LNClient.(*tests.MockLn).Pubkey = "03cbd788f5b22bd56e2714bff756372d2293504c064e03250ed16a4dd80ad70e2c"
+
+	svc.DB.Create(&db.Transaction{
+		State:          constants.TRANSACTION_STATE_FAILED,
+		Type:           constants.TRANSACTION_TYPE_OUTGOING,
+		PaymentRequest: tests.MockInvoice,
+		PaymentHash:    tests.MockPaymentHash,
+		AmountMsat:     123000,
+	})
+
+	transactionsService := NewTransactionsService(svc.DB, svc.EventPublisher)
+	transaction, err := transactionsService.SendPaymentSync(ctx, tests.MockInvoice, nil, nil, svc.LNClient, nil, nil, nil)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, transaction)
+	assert.False(t, transaction.SelfPayment)
+	assert.Equal(t, constants.TRANSACTION_STATE_SETTLED, transaction.State)
+}
+
+func TestSendPaymentSync_SelfPaymentDetection_DatabaseError(t *testing.T) {
+	ctx := context.TODO()
+
+	svc, err := tests.CreateTestService(t)
+	require.NoError(t, err)
+	defer svc.Remove()
+
+	svc.LNClient.(*tests.MockLn).Pubkey = "03cbd788f5b22bd56e2714bff756372d2293504c064e03250ed16a4dd80ad70e2c"
+
+	sqlDB, err := svc.DB.DB()
+	require.NoError(t, err)
+	sqlDB.Close()
+
+	transactionsService := NewTransactionsService(svc.DB, svc.EventPublisher)
+	transaction, err := transactionsService.SendPaymentSync(ctx, tests.MockInvoice, nil, nil, svc.LNClient, nil, nil, nil)
+
+	assert.Error(t, err)
+	assert.Nil(t, transaction)
+}
+
+func TestSendPaymentSync_SelfPaymentDetection_WithApp(t *testing.T) {
```

Comment:
What is the purpose of this test? can we remove it? (does it matter if there's an app or not? the code path is the same?)